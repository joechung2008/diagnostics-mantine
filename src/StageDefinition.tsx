import { Table, Text } from "@mantine/core";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const items = Object.entries(stageDefinition).reduce<
    KeyValuePair<string[]>[]
  >((previous, [key, value]) => [...previous, { key, value }], []);

  return (
    <div>
      <Text component="h3" size="lg" fw="semibold">
        Stage Definitions
      </Text>
      <Table>
        <Table.Thead>
          <Table.Tr>
            <Table.Th>Key</Table.Th>
            <Table.Th>Value</Table.Th>
          </Table.Tr>
        </Table.Thead>
        <Table.Tbody>
          {items.map((item) => (
            <Table.Tr key={item.key}>
              <Table.Td>{item.key}</Table.Td>
              <Table.Td>{item.value.join(", ")}</Table.Td>
            </Table.Tr>
          ))}
        </Table.Tbody>
      </Table>
    </div>
  );
};

export default StageDefinition;
