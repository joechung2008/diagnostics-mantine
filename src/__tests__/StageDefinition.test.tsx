import { describe, expect, it } from "vitest";
import StageDefinition from "../StageDefinition";
import { render } from "./test-utils";

describe("StageDefinition", () => {
  it("should render the stage definition in a table", () => {
    const stageDefinition = { stage1: ["step1", "step2"], stage2: ["step3"] };
    const { container } = render(
      <StageDefinition stageDefinition={stageDefinition} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
