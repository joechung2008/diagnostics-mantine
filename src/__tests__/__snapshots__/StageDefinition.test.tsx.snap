// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageDefinition > should render the stage definition in a table 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;
